### DISCUSSION TITLE
Too few arguments to function PermissionMiddleware::handle(), 2 passed and 3 expected

### DISCUSSION THREAD

**Discussion by irpcpro** [spatie/laravel-permission#2732] [Q&A] [answered]
Hi.
i have a question
does it support the laravel 11 ?
i have many problems for using

**Comment by irpcpro** [comment]
i'm using version 6.9 on laravel 11
but i have problems on statring.
when i add middlewares class on boorstrap/app.php i have this error :

`Too few arguments to function Spatie\Permission\Middleware\PermissionMiddleware::handle(), 2 passed in D:\Projects\ej\project\ej\vendor\laravel\framework\src\Illuminate\Pipeline\Pipeline.php on line 183 and at least 3 expected `

**Reply by parallels999** [reply]
To me, on some place you are using `PermissionMiddleware` or it alias, 
and you are not sending the name of the permission,
in Laravel 11 everything is working for me

**Reply by irpcpro** [reply, marked as answer]
thanks ..
yes after i tried i found the solution like this on `bootstrap/app.php` :
```
    ->withMiddleware(function (Middleware $middleware) {
        $middleware->alias([
            'role' => Spatie\Permission\Middleware\RoleMiddleware::class,
            'permission' => Spatie\Permission\Middleware\PermissionMiddleware::class,
            'role_or_permission' => Spatie\Permission\Middleware\RoleOrPermissionMiddleware::class,
        ]);
    })
```

**Reply by houssamHADELAMOUS** [reply]
thank you brother
